src/main/java/net/snowflake/client/core/SFBaseSession.java

```diff
@@ -717,6 +728,14 @@ public void clearSqlWarnings() {
     sqlWarnings.clear();
   }
 
+  public SFConnectionHandler getSfConnectionHandler() {
+    return sfConnectionHandler;
+  }
+
+  public void setSfConnectionHandler(SFConnectionHandler sfConnectionHandler) {
+    this.sfConnectionHandler = sfConnectionHandler;
+  }
```

Comment:
Hmm.. why a setter  ? I thought a `SFBaseSession` cannot be constructed without the connection handler. It seems wrong that the connectionHandler can be overwritten for an existing session.

Reply:
That's right. Originally, because I didn't want to mess with the constructor of `SFSession`, since it's used in a ton of tests and places. But yeah, there is a better way to do this. I'll create a new constructor with the different signature and have that "self-set" the connection handler.